addons/mrp/models/mrp_production.py

```diff
@@ -768,8 +774,8 @@ def _plan_workorders(self):
             vals['leave_id'] = leave.id
             workorder.write(vals)
         self.with_context(force_date=True).write({
-            'date_planned_start': self.workorder_ids[0].date_planned_start,
-            'date_planned_finished': self.workorder_ids[-1].date_planned_finished
+            'date_planned_start': workorder_ids[0].date_planned_start,
+            'date_planned_finished': workorder_ids[-1].date_planned_finished
```

Comment:
w1-w2-w3-w4
of replan concerns w2 and w3 you're changning date_planned_start potentially for nothing
the old code will still work